molgenis-api-data/src/test/java/org/molgenis/api/data/v3/EntityControllerIT.java

```diff
@@ -181,10 +196,33 @@ public void deleteResource() {
     given().get("/api/entity/v3_MyDataset/25").then().statusCode(NOT_FOUND.value());
   }
 
-  // TODO implement
   @Test
-  public void deleteResourceCollection() {
-    throw new UnsupportedOperationException();
+  public void deleteResourceCollection() throws IOException {
+    given().delete("/api/entity/v3_MyDataset").then().statusCode(NO_CONTENT.value());
+
+    String expectedJson = ResourceUtils.getString(getClass(), "deleteResourceCollection.json");
+
+    given()
+        .get("/api/entity/v3_MyDataset")
+        .then()
+        .statusCode(HttpStatus.OK.value())
+        .body(isEqualJson(expectedJson));
+  }
+
+  @Test
+  public void deleteResourceCollectionQuery() throws IOException {
+    given()
+        .delete("/api/entity/v3_MyDataset?q=label=in=('Row 1','Row 2','Row 3','Row 4')")
+        .then()
+        .statusCode(NO_CONTENT.value());
+
+    String expectedJson = ResourceUtils.getString(getClass(), "deleteResourceCollectionQuery.json");
+
+    given()
+        .get("/api/entity/v3_MyDataset")
```

Comment:
How about filtering on id attribute only and update the response json to make this test more resilient to changes in the data?